systemtests/src/main/java/io/enmasse/systemtest/iot/IoTUtils.java

```diff
@@ -0,0 +1,306 @@
+    private static final String IOT_LORAWAN_ADAPTER = "iot-lorawan-adapter";
+    private static final String IOT_HTTP_ADAPTER = "iot-http-adapter";
+    private static final String IOT_AUTH_SERVICE = "iot-auth-service";
+    private static final String IOT_DEVICE_REGISTRY = "iot-device-registry";
+    private static final String IOT_DEVICE_REGISTRY_MANAGEMENT = "iot-device-registry-management";
+    private static final String IOT_DEVICE_CONNECTION = "iot-device-connection";
+    private static final String IOT_TENANT_SERVICE = "iot-tenant-service";
+    private static final String IOT_SERVICE_MESH = "iot-service-mesh";
+
+    private static final Map<String, String> IOT_LABELS = Map.of("component", "iot");
+    private static final Logger log = CustomLogger.getLogger();
+
+    public static void waitForIoTConfigReady(IoTConfig config) throws Exception {
+        var budget = new TimeoutBudget(15, TimeUnit.MINUTES);
+        var iotConfigAccess = iotConfigs(config.getMetadata().getNamespace()).withName(config.getMetadata().getName());
+        TestUtils.waitUntilCondition(() -> {
+                    var currentState = iotConfigAccess.get();
+                    var currentPhase = Optional.ofNullable(currentState)
+                            .map(IoTConfig::getStatus)
+                            .map(IoTConfigStatus::getPhase)
+                            .orElse(null);
+                    if ("Active".equals(currentPhase)) {
+                        log.info("IoTConfig is ready - phase: {} -> {}", currentPhase, Serialization.asJson(currentState.getStatus()));
+                        return true;
+                    } else {
+                        log.info("Waiting until IoTConfig: '{}' will be in ready state", config.getMetadata().getName());
+                        return false;
+                    }
+                },
+                budget.remaining(), Duration.ofSeconds(5),
+                explain(() -> Optional.ofNullable(iotConfigAccess.get())
+                        .map(IoTConfig::getStatus)
+                        .map(Serialization::asJson)
+                        .orElse("IoTConfig has no status section")));
+
+        final String[] expectedDeployments = getExpectedDeploymentsNames(config);
+        final String[] expectedStatefulSets = new String[]{IOT_SERVICE_MESH};
+
+        final int meshReplicas = Optional.of(config)
+                .map(IoTConfig::getSpec)
+                .map(IoTConfigSpec::getMesh)
+                .map(MeshConfig::getServiceConfig)
+                .map(ServiceConfig::getReplicas)
+                .orElse(1);
+
+        TestUtils.waitUntilCondition("IoT Config to deploy", (phase) -> allThingsReadyPresent(expectedDeployments, expectedStatefulSets), budget);
+        TestUtils.waitForNReplicas(expectedDeployments.length + meshReplicas, config.getMetadata().getNamespace(), IOT_LABELS, budget);
+    }
+
+    public static void deleteIoTConfigAndWait(IoTConfig config) throws Exception {
```

Comment:
As follow up to my comment. I think for example this method should be moved to resource manager with created proper resources

Reply:
resource manager is not here for storing methods like before, if you want to proper use new resource manager approach you need to create <CR>ResouceType where CR is name of resource like we did for all shared infra resources and implement common methods.

Reply:
That is what I meant, sorry for my bad naming. I thought that with creation of proper Resource type which is afterwards handled by ResourceManager such as it is now for all other tests..